talksapp/main_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd.
+
+package talksapp
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/gddo/gosrc"
+
+	"appengine"
+	"appengine/aetest"
+	"appengine/memcache"
+)
+
+const importPath = "github.com/user/repo/path/to/presentation.slide"
+
+func TestHome(t *testing.T) {
+	i, _ := aetest.NewInstance(nil)
+	defer i.Close()
+
+	r, _ := i.NewRequest("GET", "/", nil)
+	w := httptest.NewRecorder()
+
+	handlerFunc(serveRoot).ServeHTTP(w, r)
+
+	if w.Code != http.StatusOK {
+		t.Fatalf("expected status: %d, got: %d", http.StatusOK, w.Code)
+	}
+
+	if !strings.Contains(w.Body.String(), "go-talks.appspot.org") {
+		t.Fatal("expected response to contain: go-talks.appspot.org")
+	}
+}
+
+func TestPresentation(t *testing.T) {
+	presentationSrc := `
+My awesome presentation!
+
+Subtitle
+
+* Slide 1
+
+- Foo
+- Bar
+- Baz
+`
+	getPresentation = func(client *http.Client, importPath string) (*gosrc.Presentation, error) {
+		return &gosrc.Presentation{
+			Filename: "presentation.slide",
+			Files: map[string][]byte{
+				"presentation.slide": []byte(presentationSrc),
+			},
+		}, nil
+	}
+
+	i, _ := aetest.NewInstance(nil)
+	defer i.Close()
+
+	r, _ := i.NewRequest("GET", "/"+importPath, nil)
+	w := httptest.NewRecorder()
+
+	handlerFunc(serveRoot).ServeHTTP(w, r)
+
+	if w.Code != http.StatusOK {
+		t.Fatalf("expected status: %d, got: %d", http.StatusOK, w.Code)
+	}
+
+	if !strings.Contains(w.Body.String(), "My awesome presentation") {
```

Comment:
Should "My awesome presentation" be a variable to factor it out of this and the definition of presentationSrc?